src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,113 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const newMovieInputs = (value: string, key: string) => {
+    setNewMovie({
+      ...newMovie,
+      [key]: value,
+    });
+  };
```

Comment:
1. function name should start with a verb
2. and add a type to `e`
```suggestion
  const addNewMovie = (e: ...) => {
    const { name, value } = e.target;
    
    setNewMovie({
      ...newMovie,
      [name]: value,
    });
  };
```